src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -361,6 +361,70 @@ def update_ag_http_listener(cmd, instance, parent, item_name, frontend_ip=None,
     return parent
 
 
+def assign_ag_identity(cmd, resource_group_name, application_gateway_name,
+                       no_wait=False, identity_type=None, user_assigned_identity=None,
+                       client_id=None, principal_id=None):
+    ncf = network_client_factory(cmd.cli_ctx).application_gateways
+    ag = ncf.get(resource_group_name, application_gateway_name)
+    ManagedServiceIdentity, ManagedServiceIdentityUserAssignedIdentitiesValue = \
+        cmd.get_models('ManagedServiceIdentity', 'ManagedServiceIdentityUserAssignedIdentitiesValue')
+    user_assigned_indentity_instance = ManagedServiceIdentityUserAssignedIdentitiesValue(
+        client_id=client_id,
+        principal_id=principal_id
+    )
+    user_assigned_identities_instance = dict()
+    if user_assigned_identity is not None:
+        user_assigned_identities_instance[user_assigned_identity] = user_assigned_indentity_instance
```

Comment:
If user_assigned_identity is not none, you will update the existing one, right? If you want to update, I think a update command may be better. Because a single resource can utilize multiple user assigned managed identities, you can assign identities many times.

Reply:
Actually, I used to have set/add/remove commands. However, I confirm with service team, now, the application-gateway only supports one identity even though its type is list. It's quite tricky. That's the reason now we only have assign and remove commands.